zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,144 @@
+2. Set up a Zulip bot user and download its `.zuliprc`
+   config file to e.g. `~/.zuliprc`
+3. Subscribe the bot to the stream that will receive Confluence updates
+   (default stream: confluence)
+4. Test the script by running it manually, like this:
+5. Configure a crontab entry for this script. A sample crontab entry
+that will process spaces and pages every 5 minutes is:
+*/5 * * * * /usr/local/share/zulip/integrations/confluence/confluence [options]
+
+== Setting up Confluence authentications ==
+
+Run this on a personal or trusted machine, because your API key is
+visible to local users through the command line or config file.
+
+This bot uses OAuth to authenticate with Confluence. Please create a
+~/.zulip_confluencerc with the following contents:
+
+[Confluence_api]
+username =
+consumer_key =
+domain_Url =
+config_file=
+
+
+In order to obtain a consumer key & secret, you must register a
+new application under your Confluence account:
+
+1 Log in to https://id.atlassian.com.
+2 Click API tokens, then Create API token.
+3 Use Copy to clipboard, and paste the token to your script
+  or elsewhere:
+
+"""
+parser = zulip.add_default_arguments(argparse.ArgumentParser("Fetch space and page from Confluence."))
+parser.add_argument('--instructions',
+                    action='store_true',
+                    help='Show instructions for the confluence bot setup and exit'
+                    )
+opts = parser.parse_args()
+if opts.instructions:
+    print(INSTRUCTIONS)
+    sys.exit()
+
+try:
+    config = ConfigParser()
+    config.read(CONFIGFILE)
+    username = config.get('Confluence_api', 'username')
+    consumer_key = config.get('Confluence_api', 'consumer_key')
+    domain_Url = config.get('Confluence_api', 'domain_Url')
+    local_file = config.get('Confluence_api', 'local_file')
+except:
+    parser.error("Please provide a ~/.zulip_confluencerc")
+
+f = open(local_file, "w+")
+f.write('0')
+f.close()
+
+def confluence_api() -> None:
+
+    params = ('type', 'page')
+    base_url = 'https://{}/wiki/rest/api/content'.format(domain_Url)
+    auth = (username, consumer_key)
+    response = requests.get(base_url, params=params, auth=auth)
+    res = response.json()
+    # length of json data
+    length_api = len(res['results'])
+    # Fetch value from database models
+    f = open(local_file, "r")
+    length_confluence_data = int(f.read())
+    # compare the lenth of json from database to the new 
+    #length of json which is fetch from api
+    # if length of fetch json from api  is greater than 
+    #save length then we call zulip api
+    if (length_api > length_confluence_data):
+        test1_json = (res['results'][length_api-1]['_links']['webui'])
+        title = res['results'][length_api-1]['title']
+        get_spaces_and_pages = spacepage(test1_json)
+        format_page_and_string = "New {} is created with title {}".format(get_spaces_and_pages, title)
+        client = zulip.Client(config_file ="/path/to/.zuliprc")
```

Comment:
The user will still have to edit this file and change "/path/to/.zuliprc" to the path to their .zuliprc. I think it would be better if  --config-file argument is used. zulip.add_default_argument sets it up so you just have to get the value from opts

Reply:
How I get the path if I use --config file argument ?

Reply:
The vaule of --config-file argument is stored in opts.zulip_config_file
https://github.com/zulip/python-zulip-api/blob/0e8347d44096d8a93d67ceb632c465d1479328f3/zulip/zulip/__init__.py#L137